### PR TITLE
Add junit formatter, bump resolvers, add nix flake

### PR DESCRIPTION
**Pull request by chris-martin**
Miscellaneous updates

### CODE REVIEW

File: tests/SpecHook.hs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  ( hook
+  ) where
+
+import Prelude
+
+import Test.Hspec
+import Test.Hspec.JUnit.Formatter.Env as Formatter
+
+hook :: Spec -> Spec
+hook = Formatter.whenEnabled Formatter.add . parallel
```
Comment (pbrisbin) [inline tests/SpecHook.hs:11]: :ok_hand:

**Review by pbrisbin — APPROVED** [review]

File: tests/SpecHook.hs (hunk 1 above)
Comment (chris-martin) [reply tests/SpecHook.hs:11]: Unfortunately it requires at minimum LTS 20.

Comment (chris-martin) [reply tests/SpecHook.hs:11]: Nevermind, just required https://github.com/freckle/hspec-junit-formatter/pull/30

**Comment by chris-martin** [comment]
It seems that adding a spec file is necessary to prevent weeder from complaining that `SpecHook` is unused.

**Pull request merged**